src/backend/joanie/core/factories.py

```diff
@@ -677,6 +682,208 @@ def main_invoice(self, create, extracted, **kwargs):
         return None
 
 
+class OrderGeneratorFactory(factory.django.DjangoModelFactory):
+    """A factory to create an Order"""
+
+    class Meta:
+        model = models.Order
+
+    product = factory.SubFactory(ProductFactory)
+    course = factory.LazyAttribute(lambda o: o.product.courses.order_by("?").first())
```

Comment:
I don't know the magic behind `order_by("?")`. Can you explain ?

Reply:
It is from the previous factory.
I believe it is to get a random ordering.